scripts/source-maintenance.sh

```diff
@@ -84,27 +101,37 @@ done
 # an error code seen by a KeepGoing-aware command (or zero)
 SeenErrors=0
 
-
 if ! git diff --quiet; then
 	echo "There are unstaged changes. This script may modify sources."
 	echo "Stage changes to avoid permanent losses when things go bad."
 	exit 1
 fi
 
-# On squid-cache.org we have to use the python scripted md5sum
-HOST=`hostname`
-if test "$HOST" = "squid-cache.org" ; then
-	MD5="md5"
+for Checksum in md5sum md5 shasum sha1sum
+do
+    if "$Checksum" </dev/null >/dev/null 2>/dev/null ; then
+        break
+    fi
+done
+echo "detected checksum program $Checksum"
+
```

Comment:
The above will leave Checksum set to sha1sum when sha1sum is known not to work. Should not we _quit_ with an error when all tried spellings failed?

Reply:
Good idea